src/api/update-issue.mutation.ts

```diff
@@ -0,0 +1,80 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import type { z } from "zod";
+import { log } from "../lib/log.js";
+import type { issue } from "./issue-query.js";
+import { request } from "./request.js";
+
+interface UpdateIssueMutationVariables {
+  issueId: string;
+  fields: Record<string, unknown>;
+}
+
+const updateIssue = async (variables: UpdateIssueMutationVariables) => {
+  await request(`api/3/issue/${variables.issueId}`, {
+    method: "PUT",
+    body: JSON.stringify({
+      fields: variables.fields,
+    }),
+  });
+};
+
+export const useUpdateIssueMutation = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation<
+    unknown,
+    Error,
+    UpdateIssueMutationVariables,
+    { previousIssues: z.infer<typeof issue>[] }
+  >({
+    mutationFn: updateIssue,
+    onMutate: (variables) => {
+      queryClient.cancelQueries({ queryKey: ["issues"] });
+
+      const issues = queryClient.getQueryData(["issues"]) as z.infer<
+        typeof issue
+      >[];
+
+      queryClient.setQueryData(
+        ["issues"],
+        issues.map((issue) => {
+          if (issue.id === variables.issueId) {
+            const updatedFields: Record<string, unknown> = {};
+
+            for (const [key, value] of Object.entries(variables.fields)) {
+              if (typeof value === "object") {
```

Comment:
Unsure if it matters, but arrays are objects, don't know if there's an instance where you could be overwriting it with an object now instead of an array